### ISSUE TITLE
Route Claim | Display Informational Route Claim unable to create EP (Full Grants with Special Issues/RO Email)

### ISSUE THREAD

**Issue by lakohl** [department-of-veterans-affairs/caseflow#722]
As a Caseworker, when I am working a case that is a Full Grant with a special issue for whom we cannot create an EP in Caseflow but for whom we do have an RO/Center of Jurisdiction email address available, I would like to have message displayed so that I am aware of the situation.

# Acceptance Criteria
1. Verify that after clicking the **_Route Claim_** button on the `Decision Review` page and the case is a Full Grant with unsupported Special Issues and contains an RO Email, the `Route Claim` page is displayed with the following Title, Informational Alert message, and subtext:  (Refer to Mockup)

     **Title**

     > Route Claim

     **Informational Message**

     > We are unable to create an EP for claims with this Special Issue.
     > Follow the instructions below to route this claim.

     **Subtext**

     > Please send the following email message to the office responsible for implementing this grant.

1. Verify that on the `Route Claim` modal when there is an RO/Center of Jurisdiction email available, the the following data elements are displayed (refer to mockup):

     - **RO** - Regional Office's code concatenated with its city and state.  _Display Only._
          _Format_:  [RO_Code] - [RO_City], [RO_State]
     - **RO Email** - Regional Office or Center of Jurisdiction email address.  _Display Only_.
     - **Message** - pre-filled email verbiage that will be copied/pasted into an email for sending to the RO.  _Ediitable_.
     - **I confirm that I have sent an email to route this claim.** - confirmation of email being sent.

1. Verify that the _"Message"_ field contains the following message: 

     > The BVA Full Grant decision dated [Decision_Date] for [Veteran_FirstName] [Veteran_LastName], ID [Veteran_ID], was sent to the ARC but cannot be processed here, as it contains [Special_Issue_Category] in your jurisdiction. Please proceed with control and implementation of this grant.

1. Verify that the following links are available: 
     - **_Back to Decision Review_** 
     - **_Cancel_**
1. Verify that the following button is available:  **_Finish Routing Claim_**
1. Verify that when "I confirm that I have sent an email to route this claim." has been checked and there is an RO Email, the **_Finish Routing Claim_** button has been enabled.
1. Verify that after clicking on the **_Finish Routing Claim_** button, the `EP Confirmation` page is displayed with the following checklist items (refer to mockup):

     - Reviewed Decision
     - Sent Email Notification

1. Verify that after clicking on the **_Cancel_** link, the `Stop Processing Claim` modal is displayed.
1. Verify that after clicking on the **_Back to Decision Review_** link, the `Caseworker Dashboard` page is displayed.
## Mockups
_**View - Route Claim - Informational Alert (RO Email Available)**_
<img width="770" alt="screen shot 2017-02-13 at 5 16 33 pm" src="https://cloud.githubusercontent.com/assets/4975959/22905947/49413654-f210-11e6-8039-7236b6239da9.png">

_**View - Route Claim - Information Alert (RO Email Available) - Finish Routing Claim button enabled**_
<img width="748" alt="screen shot 2017-02-13 at 5 16 39 pm" src="https://cloud.githubusercontent.com/assets/4975959/22905953/4d12651e-f210-11e6-8e69-25fb94077042.png">

_**View - EP Confirmation**_
<img width="584" alt="screen shot 2017-02-13 at 12 55 58 pm" src="https://cloud.githubusercontent.com/assets/4975959/22895817/cb574126-f1eb-11e6-9605-4178a86c0d19.png">

_**View - Stop Processing Claim**_ 
![stop processing claim](https://cloud.githubusercontent.com/assets/23063044/23077115/2eda984c-f511-11e6-80ce-6dda2238ca74.PNG)

# Related Stories
Route Claim | Display Alert Route Claim unable to create EP (Full Grants w/ Special Issue, & no RO Email)  #957

**Comment by lakohl** [comment]
@abbyraskin I took a stab at the copy for this, would you mind taking a look?

'You selected a Full Grant with a Special Issue Category not handled by AMO. This type of case cannot be processed in Caseflow at this time.

Please select Cancel Claim Establishment and send the following email to the RO associated with this case in VACOLS.

RO Email: specialissuesStPeteRO@va.gov

Message: Claim #123456789 was sent to AMO and we are unable to work it here. It has a full grant decision with [special issue category] in your jurisdiction. Please proceed with the adjustment of this claim.'

<img width="540" alt="screen shot 2017-01-26 at 8 49 49 am" src="https://cloud.githubusercontent.com/assets/4975959/22333487/71ee52aa-e3a4-11e6-816b-9d50e38f3416.png">

**Comment by lakohl** [comment]
@abbyraskin next we'll have to work on determining routing/email addresses for all full grant special issues : 0

**Comment by abbyraskin** [comment]
@lakohl is it possible to actually generate the email for them (in their own Outlook, even though we don't actually send the email through our system)?

**Comment by abbyraskin** [comment]
I would also include the date and claimant's name in the email. 

The BVA decision dated 01/01/2017 for John Doe, ID 123456789, was sent to the AMO but cannot be processed here, as it contains [special issue category] in your jurisdiction. Please proceed with implementing this grant.

**Comment by abbyraskin** [comment]
I would also include the date and claimant's name in the email.

The BVA decision dated 01/01/2017 for John Doe, ID 123456789, was sent to the AMO but cannot be processed here, as it contains [special issue category] in your jurisdiction. Please proceed with control and implementation of this claim.

**Comment by lakohl** [comment]
Good point about the name.
Last I heard the dev team wanted to do this in incremental steps. Generating the email in our system would come later I think Outlook link may be doable now though. Will double check

**Comment by lakohl** [comment]
Hold for update.

**Comment by shanear** [comment]
@lakohl to add what is shown, and when?

**Comment by shanear** [comment]
"I have sent the email" should be the button text

**Comment by lakohl** [comment]
@shanear when we talked yesterday I completely forgot that Abby had another - good idea for handling this. Same functionality, but instead of asking the user to email the RO, in the cases where we create and EP we would ask the user to go into VBMS and manually add a BGS note. This would get them in the habit of using the BGS notes for transferring Full Grants and would give us more of a trail. Eventually we would populate the note for them.   
For cases where there is no EP, we would still ask them to email.

**Comment by lakohl** [comment]
@shanear I'm still working on this, but it's now less of a priority than https://github.com/department-of-veterans-affairs/caseflow/issues/833

**Comment by lakohl** [comment]
Possible solution:
![screen shot 2017-02-08 at 12 08 46 am](https://cloud.githubusercontent.com/assets/4975959/22724699/f60bac36-ed95-11e6-986e-70f56e214f35.png)

![screen shot 2017-02-08 at 12 08 54 am](https://cloud.githubusercontent.com/assets/4975959/22724702/ff5fecde-ed95-11e6-9403-c6848dc01788.png)

![screen shot 2017-02-08 at 12 10 55 am](https://cloud.githubusercontent.com/assets/4975959/22724705/08b8c170-ed96-11e6-9cb6-f1a70d63b7b4.png)

**Comment by lakohl** [comment]
List of RO emails is being collected here: https://github.com/department-of-veterans-affairs/appeals-design-research/issues/22#issuecomment-278786304

**Comment by lakohl** [comment]
@abbyraskin and @avrildavid Would you mind taking a look at the copy/styling of the last two mocks yellow warning at the top of this issues. They serve the function of a general release claim message for Full Grants with Special Issues not handled in Caseflow for which we have not yet located email addresses.

**Comment by lakohl** [comment]
@sharonwarner I just updated the AC at the top to reflect our conversation this am. Am waiting on feedback on copy fro the more general modal that is served up when we don't have an email address.

**Comment by abbyraskin** [comment]
I would reword "Send the following email message to the RO associated with this case in VACOLS" because a) these won't apply to ROs -- specialized centers + NCA/VHA b) the "correct" destination may not be listed on the appeal in VACOLS if there was a hearing.

Maybe..."Send the following email message to office responsible for implementing this grant."

**Comment by avrildavid** [comment]
The above makes sense to me.

**Comment by lakohl** [comment]
@abbyraskin @avrildavid thanks for the feedback. I realized that the mocks with yellow warning messages didn't 'stick' earlier. Would you mind taking a look at those too?

**Comment by abbyraskin** [comment]
@lakohl I think the ordering in the instructions at the end ("Please select Release Claim and proceed to process this case manually") is a little confusing, given that they can't actually select the button until they've checked a box saying they processed the case - what if we said something like “Please process this case manually and select Release Claim when you are finished" instead?

**Comment by lakohl** [comment]
Thanks @abbyraskin

**Comment by lakohl** [comment]
@sharonwarner copy and AC mocks (with yellow warning bar) have been updated to include @abbyraskin's most recent comments. Thanks for your patience!

**Comment by sharonwarner** [comment]
@lakohl @abbyraskin do we need to save the special issues in the database after completing the task?

**Comment by lakohl** [comment]
@sharonwarner yes. when in doubt, always save the special issues.

**Comment by lakohl** [comment]
@sharonwarner Abby created a table with all the RO email addresses here: https://github.com/department-of-veterans-affairs/appeals-design-research/blob/master/Projects/Dispatch/Routing%20EPs/Email%20Routing%20Table.md

**Comment by kierachell** [comment]
There's no _Required_ in the mocks.
And if it should be there, probably not like this:
![noterequired](https://cloud.githubusercontent.com/assets/18075411/23188640/798811a4-f85d-11e6-986c-27ee13e6b94c.png)

**Comment by lakohl** [comment]
also, in production hopefully the message will list only the relevant special issue(s) not all that apply

**Comment by lakohl** [comment]
@sharonwarner @kierachell I updated my comment above in case the wording was confusing

**Comment by kierachell** [comment]
![rocodenocity](https://cloud.githubusercontent.com/assets/18075411/23193860/fd7addfa-f878-11e6-8f14-1e2eb85cc4f6.png)
 : do we have an up-to-date list of RO Codes - Cities?
@lakohl @abbyraskin 
![specialissuenote](https://cloud.githubusercontent.com/assets/18075411/23193903/1bc17ab2-f879-11e6-9703-30505065d658.png)

**Comment by abbyraskin** [comment]
@kierachell I have [this list](https://github.com/department-of-veterans-affairs/appeals-design-research/blob/master/Projects/Dispatch/Routing%20EPs/Station%20Codes.csv).

**Comment by kierachell** [comment]
Currently I see no City/State shown for any of the RO codes

**Comment by kierachell** [comment]
**PASSED**
All appropriate ROs display Cities/States.
At least the ones we know.
